Dolphin scripts/Entrance Randomizer/lib/graph_creation.py

```diff
@@ -54,7 +82,7 @@ def create_vertices(
         # The same logic applies to the Spirit Fights:
         # these will never appear on the map, therefore we remove the (Harry) suffix.
         area_name = (
-            TRANSITION_INFOS_DICT
+            transition_infos_dict_rando
```

Comment:
To name one thing it allows to display `Burning Outpost` and `Jungle Outpost`, both of which aren't present in `TRANSITION_INFOS_DICT`, as that dict considers them both to be equal to `Twin Outposts`.

Reply:
Hmm... I'd rather not have the graph *depend* on `entrance_rando` (by separation of concerns, the main script *uses* the transition rando results as an input for the graph, and will do so for item rando, custom shaman shop, etc.). Before making any change, that brings up a design question:
Should the outposts be seen as different levels?
It's possible to traverse across with dash or pickaxe, meaning if they're seen as different levels, they should always be connected together on the graph. This question will also be relevant long term with item logic.
I also feel like the outposts only make sense being seen as different levels if the wells are also randomized (which we can't do rn either, because the "previous area" in-memory data isn't sufficient to spawn onto the right well (although there is a hacky workaround which is to simply *not* spawn on the well, and use the other 2 entrances instead)

Reply:
> To name one thing it allows to display `Burning Outpost` and `Jungle Outpost`, both of which aren't present in `TRANSITION_INFOS_DICT`, as that dict considers them both to be equal to `Twin Outposts`.

That's not implemented yet though, right?
![image](https://github.com/user-attachments/assets/a5f90808-45eb-4402-85ef-ed8781bd39fc)

Reply:
- Yes the outposts should be seen as different levels, because if you don't you cause all kinds of problems + it's not intuitive at all for players that these are the same level, as it's not possible at all to go from the Flooded Courtyard exit to the Turtle Monument exit without using very specific glitches.
- If you as a player are crazy enough to traverse between the two using an item then I say good for you. This shouldn't impact the map generation whatsoever. The furthest I'm willing to go here is add an extra dotted line between the levels after all other generation is done, and even that's a stretch.
- I don't know what you mean by randomizing wells. What "wells"?
- About it not being implemented yet: I already have it implemented on my local version for weeks now. The only reason you don't have it yet is because you insist on smaller merge requests, and the merge requests we've done so far have been so slow.

So long story short: I thought about all of this and doing it this way was a very well thought out choice.

Reply:
By "wells" I mean the entrances to the underwater area (the entrance is inside of a well on both sides, that's what I was referring too).

I guess they could always be marked separately on the graph with "unrandomized" connection in between, or w/e visual marker, now that that's a thing. So less of a concern.

> About it not being implemented yet: I already have it implemented on my local version

Ok I thought I missed something in this PR.

To answer your question on Discord about fixing the newly appeared pyright issue due to the name containing an underscore:
It simply shows in a more obvious way the "leak" of responsibility between `entrance_rando` and `graph_creation`.
1. You *could* do the necessary code modification to solve that with in mind whatever other changed you have prepped that I'm not necessarily aware of. But neither you nor I would like to add that to this PR.
2. You could go back to using `TRANSITION_INFOS_DICT`, which currently has the exact same effect, reduces changes here (ie: the things you mentioned about Twin Outpost is not of concern for this PR, but is good to know for later). And the need for this data can be addressed with the PR that adds it.
3. You can add `# pyright: ignore [theErrorCode]` to get this PR green, and I'll just revert it to `TRANSITION_INFOS_DICT` after anyway, so like 2. the needs for this data can be addressed in the proper PR.

Reply:
By "wells" I mean the entrances to the underwater area (the entrance is inside of a well on both sides, that's what I was referring too).

I guess they could always be marked separately on the graph with "unrandomized" connection in between, or w/e visual marker, now that that's a thing. So less of a concern.

> About it not being implemented yet: I already have it implemented on my local version

Ok I thought I missed something in this PR.

To answer your question on Discord about fixing the newly appeared pyright issue due to the name containing an underscore:
It simply shows in a more obvious way the "leak" of responsibility between `entrance_rando` and `graph_creation`.
1. You *could* do the necessary code modification to solve that with in mind whatever other changed you have prepped that I'm not necessarily aware of. But neither you nor I would like to add that to this PR.
2. You could go back to using `TRANSITION_INFOS_DICT`, which currently has the exact same effect, reduces changes here (ie: the things you mentioned about Twin Outpost is not of concern for this PR, but is good to know for later). And the need for this data can be addressed with the PR that adds it.
3. You can add `# pyright: ignore [theErrorCode]` to get this PR green, and I'll just revert it to `TRANSITION_INFOS_DICT` after anyway, so like 2. the needs for this data can be addressed in the proper PR.